### DISCUSSION TITLE
My contributions are not showing on graph

### DISCUSSION THREAD

**Discussion by hammadronaldo** [community/community#57441] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

idk why my commits aren't showing as the contributions on the graph. :( I dont know the solution.

**Comment by dakshdeepHERE** [comment, marked as answer]
**This happened with me too and if your mail is right on your local device and your GitHub then there can be 3 issues/points you have to consider:**
- First you might be sending commits to wrong remote repo you can confirm it by using the command **`git remote -v`** it will return the link and you can confirm the repo you are trying to contribute in.
- Second you might have forked a repo and contributing on the forked repo which again won't show on the contribution graph until your forked repo don't get merged into the active branch of the real repo and there are valid reasons behind this like:
  - Collaboration vs. Independent Work: the purpose of counting contributions on a profile is to highlight the collaborative work done directly on a project. Commits in forks are often considered independent work since they are not made directly in the original repository.

  - Maintainer Control: repository maintainers have control over whether to include contributions made in forks in their project's statistics. They can choose to merge pull requests from forks and attribute the contributions to the original repository, but it's their discretion. 
  - Privacy and Security: forks provide a level of privacy and security, as they allow users to experiment, make changes, and test without directly impacting the original repository. Counting commits in forks as contributions on a profile could potentially expose sensitive work that users may not want to publicize.
- Third, if you are committing on the different branch which is not the active branch then also your commit won't be converted into contribution until you them merged with the active branch which seems to be the case here.

More over you can read this [doc](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/why-are-my-contributions-not-showing-up-on-my-profile) too.

**Comment by tavindev** [comment]
My git email was wrong for some reason, and all my past 2 weeks commits to private repositories which I own were not counted. Is it possible to somehow update these commits with the right email?

**Reply by dakshdeepHERE** [reply]
Yes there is a way. I will mention the steps below you have to follow them:
- Still in the terminal or command prompt, navigate to the repository where you want to update the commit email history.
- Run the following command, replacing old-email@example.com with the incorrect email you have been using, and new-email@example.com with the correct email:
```shell
git filter-branch --commit-filter '
    if [ "$GIT_AUTHOR_EMAIL" = "old-email@example.com" ];
    then
        GIT_AUTHOR_EMAIL="new-email@example.com";
        GIT_COMMITTER_EMAIL="new-email@example.com";
        git commit-tree "$@";
    else
        git commit-tree "$@";
    fi' HEAD
```
- This command will rewrite the commit history, replacing the incorrect email address with the correct one for all the previous commits.
- After completing these steps, your future commits will be associated with the new email address, and the previous commits will reflect the correct email address as well. However, note that rewriting commit history can have consequences, especially if the repository is shared or if you have already pushed the incorrect commits to a remote repository. But as you mentioned above it's your own private repo you good to go.
Hope I was able to help you @tavindev. Good luck!

**Reply by tavindev** [reply]
Thanks @dakshdeepHERE. It worked.

For anyone who tries this as well, if get duplicate commits, after running the command use `git push --force --tags origin HEAD:master`.

**Reply by Nozibul** [reply]
1. My contributions are not showing on graph?
2. How to show last month all the projects committed to the github repository in the contribution graph?

**Reply by Nozibul** [reply]
D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>    if [ "$GIT_AUTHOR_EMAIL" = "nozibulislamspi@gmail.com" ];
"$GIT_AUTHOR_EMAIL" was unexpected at this time.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>    then
'then' is not recognized as an internal or external command,
operable program or batch file.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>        GIT_AUTHOR_EMAIL="nozibulislamspi@gmail.com";
'GIT_AUTHOR_EMAIL' is not recognized as an internal or external command,
operable program or batch file.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>        GIT_COMMITTER_EMAIL="nozibulislamspi@gmail.com";
'GIT_COMMITTER_EMAIL' is not recognized as an internal or external command,
operable program or batch file.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>        git commit-tree "$@";
fatal: not a valid object name $@;

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>    else
'else' is not recognized as an internal or external command,
operable program or batch file.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>        git commit-tree "$@";
fatal: not a valid object name $@;

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>    fi' HEAD
'fi'' is not recognized as an internal or external command,
operable program or batch file.

D:\next.js\mosh\Issues tracker scalable web app\issue-tracker>

you will mention the steps below  have to follow me but it;s not solved.

**Reply by Nozibul** [reply]
1. My contributions are not showing on graph?
2. How to show last month all the projects committed to the github repository in the contribution graph?
